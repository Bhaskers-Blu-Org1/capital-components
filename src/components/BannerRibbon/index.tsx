import Ribbon from "./BannerRibbon";
import Breadcrumb from "./Breadcrumb";
import DropdownWrapper from "./DropdownWrapper";

export default {
  Ribbon,
  DropdownWrapper,
  Breadcrumb
};
