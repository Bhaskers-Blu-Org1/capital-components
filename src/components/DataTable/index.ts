export { default, IColumnDescriptor } from "./DataTable";
export * from "./utils";
