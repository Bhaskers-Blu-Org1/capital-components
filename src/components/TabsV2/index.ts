export { default } from "./TabsV2";
