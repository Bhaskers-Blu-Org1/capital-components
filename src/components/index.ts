export { default as Col } from "./Col";
export { default as Grid } from "./Grid";
export { default as DataTable } from "./DataTable";
export { default as FlyOver } from "./FlyOver";
export { default as FlyOverProvider } from "./FlyOverProvider";
export { default as Icon } from "./Icon";
export { default as Table } from "./Table";
export { default as Popover } from "./Popover";
export { default as DropMenu } from "./DropMenu";
export { default as DropMenuItem } from "./DropMenuItem";
export { default as Tabs } from "./Tabs";
export { default as BleedingRow } from "./BleedingRow";
export { default as PushOver } from "./PushOver";
export { default as PushOverItem } from "./PushOverItem";
export { default as TabsV2 } from "./TabsV2";
export { default as Tab } from "./Tab";
