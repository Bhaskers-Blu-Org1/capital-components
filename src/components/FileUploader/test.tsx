import { cleanup, fireEvent, getByRole as getByRoleGlobal } from "react-testing-library";
import "jest-dom/extend-expect";
import { FileUploader } from "@fss/components";
import React from "react";
import { renderWithDefaultTheme } from "test-utils";

afterEach(cleanup);

test("Basic snapshot", async () => {
  const { baseElement } = renderWithDefaultTheme(<FileUploader onFilesAdded={jest.fn()} />);

  // When not expandable, it is a header element
  expect(baseElement).toMatchSnapshot();
});

test("File added shows in list and can be removed", async () => {
  const onFilesRemoved = jest.fn();
  const initialFiles = [
    new File(["Some content"], "file1.txt"),
    new File(["Some content"], "file2.txt")
  ];
  const { getByLabelText } = renderWithDefaultTheme(
    <FileUploader
      onFilesAdded={jest.fn()}
      files={initialFiles}
      canRemoveFile={() => true}
      onFilesRemoved={onFilesRemoved}
    />
  );

  const deleteFileButton = getByLabelText("Remove file file1.txt");
  expect(deleteFileButton).toBeInTheDocument();
  fireEvent.click(deleteFileButton);

  expect(onFilesRemoved).toBeCalledTimes(1);
  expect(onFilesRemoved.mock.calls[0][0]).toHaveLength(1);
  expect(onFilesRemoved.mock.calls[0][0][0].name).toBe("file1.txt");
});

test("Support clickable elements", async () => {
  const onFileClick = jest.fn();
  const initialFiles = [
    new File(["Some content"], "file1.txt"),
    new File(["Some content"], "file2.txt")
  ];
  const { getByTitle } = renderWithDefaultTheme(
    <FileUploader
      files={initialFiles}
      onFileClick={onFileClick}
      canClickFile={() => true}
      onFilesAdded={jest.fn()}
    />
  );

  const element = getByTitle("file1.txt");
  expect(element).toBeInTheDocument();
  const filenameElement = getByRoleGlobal(element, "button");
  fireEvent.click(filenameElement);
  expect(onFileClick).toBeCalledTimes(1);
  expect(onFileClick.mock.calls[0][0].name).toBe("file1.txt");
});

test("Cannot click elements that are not enabled", async () => {
  const onChange = jest.fn();
  const onFileClick = jest.fn();
  const initialFiles = [
    new File(["Some content"], "file1.txt"),
    new File(["Some content"], "file2.txt")
  ];
  const { getByText } = renderWithDefaultTheme(
    <FileUploader
      files={initialFiles}
      onFilesAdded={onChange}
      onFileClick={onFileClick}
      canClickFile={file => file.name === "file1.txt"}
    />
  );

  const file1 = getByText("file1.txt");
  const file2 = getByText("file2.txt");

  fireEvent.click(file1);
  fireEvent.click(file2);

  expect(onFileClick).toBeCalledTimes(1);
  expect(onFileClick.mock.calls[0][0]).toMatchObject({ name: "file1.txt" });
});

test("You can disable some files from being deleted", async () => {
  const onRemove = jest.fn();
  const initialFiles = [
    new File(["Some content"], "file1.txt"),
    new File(["Some content"], "file2.txt")
  ];
  const { getByLabelText, queryByLabelText } = renderWithDefaultTheme(
    <FileUploader
      files={initialFiles}
      onFilesAdded={jest.fn()}
      onFilesRemoved={onRemove}
      canRemoveFile={file => file.name === "file1.txt"}
    />
  );

  const file1Delete = getByLabelText("Remove file file1.txt");
  expect(file1Delete).toBeInTheDocument();
  fireEvent.click(file1Delete);
  expect(onRemove).toBeCalledTimes(1);
  expect(onRemove.mock.calls[0][0]).toMatchObject([{ name: "file1.txt" }]);
  expect(onRemove.mock.calls[0][0]).toHaveLength(1);

  expect(queryByLabelText("Remove file file2.txt")).not.toBeInTheDocument();
});
