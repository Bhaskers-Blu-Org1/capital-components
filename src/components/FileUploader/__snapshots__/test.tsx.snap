// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic snapshot 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-width: 2px;
  border-color: #dcdcdc;
  background-color: #ffffff;
  border-style: dashed;
  border-radius: 2px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 1rem;
  width: 100%;
}

.emotion-0 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #0530ad;
  cursor: pointer;
}

<body>
  <div>
    <div
      class="emotion-4 emotion-5"
      tabindex="0"
    >
      <div
        class="emotion-2 emotion-3"
      >
        <input
          autocomplete="off"
          data-testid="wfss-file-uploader-input"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
        <span
          class="emotion-0 emotion-1"
        >
          Drop files to attach, or browse
        </span>
      </div>
    </div>
  </div>
</body>
`;
